### PR TITLE
fix(Triangulate): change algorithm for ear clipping

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: src/geode/mesh/helpers/convert_surface_mesh.cpp [hunk 1]
```diff
@@ -384,20 +414,50 @@ namespace geode
...
                 for( const auto e : LRange{ nb_vertices } )
                 {
-                    adjacents[e] = surface.polygon_adjacent_edge( { p, e } );
+                    if( const auto adj =
+                            surface.polygon_adjacent_edge( { p, e } ) )
+                    {
+                        adjacents.emplace(
+                            Edge{ vertices[e],
+                                vertices[e + 1 == nb_vertices ? 0 : e + 1] },
+                            adj.value() );
```
Comment (panquez) [inline src/geode/mesh/helpers/convert_surface_mesh.cpp:428]: are there always a value?

Comment (BotellaA) [reply src/geode/mesh/helpers/convert_surface_mesh.cpp:428]: No but it is protected inside an if block

**Review by panquez — APPROVED** [review]

**Pull request merged**